DSharpPlus/Entities/Interaction/DiscordInteraction.cs

```diff
@@ -67,7 +67,7 @@ public DiscordGuild Guild
         /// </summary>
         [JsonIgnore]
         public DiscordChannel Channel
-            => (this.Discord as DiscordClient).InternalGetCachedChannel(this.ChannelId);
+            => (this.Discord as DiscordClient).InternalGetCachedChannel(this.ChannelId) ?? (this.Guild == null ? new DiscordDmChannel { Id = this.ChannelId, Type = ChannelType.Private, Discord = this.Discord } : null);
```

Comment:
I don't believe we cache DM channels anymore so you can just have it create a new DM channel rather than checking if the guild is null.

Reply:
We do cache them when creating a dm channel https://github.com/DSharpPlus/DSharpPlus/blob/master/DSharpPlus/Net/Rest/DiscordApiClient.cs#L1455-L1456

Reply:
For now, however I'm planning on removing this in the future. I guess this can stay but it'll have to be reworked at some point.